api/src/utils/apply-query.ts

```diff
@@ -606,132 +607,128 @@ export function applyFilter(
-			}
+				case '_lt':
+					dbQuery[logical].where(selectionRaw, '<', compareValue);
+					break;
 
-			if (operator === '_lte') {
-				dbQuery[logical].where(selectionRaw, '<=', compareValue);
-			}
+				case '_lte':
+					dbQuery[logical].where(selectionRaw, '<=', compareValue);
+					break;
 
-			if (operator === '_in') {
-				let value = compareValue;
-				if (typeof value === 'string') value = value.split(',');
+				case '_in':
+					if (typeof compareValue === 'string') compareValue = compareValue.split(',');
 
-				dbQuery[logical].whereIn(selectionRaw, value as string[]);
-			}
+					dbQuery[logical].whereIn(selectionRaw, compareValue as string[]);
+					break;
 
-			if (operator === '_nin') {
-				let value = compareValue;
-				if (typeof value === 'string') value = value.split(',');
+				case '_nin':
+					if (typeof compareValue === 'string') compareValue = compareValue.split(',');
 
-				dbQuery[logical].whereNotIn(selectionRaw, value as string[]);
-			}
+					dbQuery[logical].whereNotIn(selectionRaw, compareValue as string[]);
+					break;
 
-			if (operator === '_between') {
-				if (compareValue.length !== 2) return;
+				case '_between':
+					if (compareValue.length !== 2) return;
 
-				let value = compareValue;
-				if (typeof value === 'string') value = value.split(',');
+					if (typeof compareValue === 'string') compareValue = compareValue.split(',');
+					dbQuery[logical].whereBetween(selectionRaw, compareValue);
+					break;
 
-				dbQuery[logical].whereBetween(selectionRaw, value);
-			}
-
-			if (operator === '_nbetween') {
-				if (compareValue.length !== 2) return;
+				case '_nbetween':
+					if (compareValue.length !== 2) return;
 
-				let value = compareValue;
-				if (typeof value === 'string') value = value.split(',');
+					if (typeof compareValue === 'string') compareValue = compareValue.split(',');
+					dbQuery[logical].whereNotBetween(selectionRaw, compareValue);
+					break;
 
-				dbQuery[logical].whereNotBetween(selectionRaw, value);
-			}
+				case '_intersects':
+					dbQuery[logical].whereRaw(helpers.st.intersects(key, compareValue));
+					break;
 
-			if (operator == '_intersects') {
-				dbQuery[logical].whereRaw(helpers.st.intersects(key, compareValue));
-			}
+				case '_nintersects':
+					dbQuery[logical].whereRaw(helpers.st.nintersects(key, compareValue));
+					break;
 
-			if (operator == '_nintersects') {
-				dbQuery[logical].whereRaw(helpers.st.nintersects(key, compareValue));
-			}
+				case '_intersects_bbox':
+					dbQuery[logical].whereRaw(helpers.st.intersects_bbox(key, compareValue));
+					break;
 
-			if (operator == '_intersects_bbox') {
-				dbQuery[logical].whereRaw(helpers.st.intersects_bbox(key, compareValue));
-			}
-
-			if (operator == '_nintersects_bbox') {
-				dbQuery[logical].whereRaw(helpers.st.nintersects_bbox(key, compareValue));
+				case '_nintersects_bbox':
+					dbQuery[logical].whereRaw(helpers.st.nintersects_bbox(key, compareValue));
+					break;
```

Comment:
Im not to sure about the unneeded refactor to switch here in this PR. This has no direct connection to the fix done and no direct advantage im aware of. If this change is to be done it deserves its own PR with justification for why.

Personally i consider this change a regression in both readability and maintainability.

Reply:
It does execute faster than having multiple ifs and I feel it is much easier to read a switch knowing that only one of the cases will execute, rather than multiple ifs being all checked.

As for a different PR, I agree it wasn't part of the original issue, but since I was busy implementing "returning early" once operator was matched, I just felt it natural to refactor the rest of the function i the same fashion.

The logic is still the same for refactored code.

Reply:
In modern browsers `switch-cases` are not faster than `if-else` in any meaningful way https://www.measurethat.net/Benchmarks/Show/17159/0/if-else-vs-switch

Reply:
If only if-else was actually used ;)

Reply:
Ok, I can't fight the data. Apparently just using IF, is some how faster :D

So I could change it back, although I think switch reads easier, vote? :)

![Screenshot 2023-01-31 at 14 48 59](https://user-images.githubusercontent.com/6641242/215777981-1633b1f0-1a3c-466c-9814-ea8cf305ec07.png)
![Screenshot 2023-01-31 at 14 49 51](https://user-images.githubusercontent.com/6641242/215778148-405de670-4934-4f6e-b5cc-d878b9c1bc84.png)